extern/storage-sealing/states_failed.go

```diff
@@ -87,6 +87,18 @@ func (m *Sealing) handlePreCommitFailed(ctx statemachine.Context, sector SectorI
 				return err
 			}
 
+			pci, pciErr := m.Api.StateSectorPreCommitInfo(ctx.Context(), m.maddr, sector.SectorNumber, tok)
+			if pciErr != nil {
+				log.Errorf("handlePreCommitFailed: StateSectorPreCommitInfo: api error, not proceeding: %+v", pciErr)
+				return nil
+			}
+
+			// PreCommitMessage is not nil, but SectorPreCommitInfo is nil, should try to resubmit
+			if pci == nil {
```

Comment:
This can also be nil after the message was submitted but wasn't yet executed on-chain (which can take 1-2min)

Reply:
PreCommitMessage is not nil, it should be possible to determine whether the status is jumped from PreCommitWait or PreCommitBatchWait to PreCommitFailed

Reply:
In `func(m *Sealing) handlePreCommitWait`, I understand that StateWaitMsg should block until query result or return err, is it so?